src/common/k8s-api/__tests__/kube-api.test.ts

```diff
@@ -179,10 +155,30 @@ describe("KubeApi", () => {
       checkPreferredVersion: true,
     });
 
+
+    fetchMock.mockImplementation(async (url) => {
```

Comment:
Nit: using `asyncFn` here would make test easier to read, as then code would be written in same order as it is ran.

Reply:
Probably it would be easier to read. I even started trying to use it but the tests are much harder to write because `await fetchMock.resolve()` throws if a call is not currently waiting for it, which is rather annoying.

Reply:
This we talked and studied somewhere else with team? Resolved :)